src/libos/src/net/socket/unix/stream/address_space.rs

```diff
@@ -0,0 +1,93 @@
+use super::endpoint::Endpoint;
+use super::stream::Listener;
+use super::*;
+use std::collections::btree_map::BTreeMap;
+
+lazy_static! {
+    pub(super) static ref ADDRESS_SPACE: AddressSpace = AddressSpace::new();
+}
+
+pub struct AddressSpace {
+    file: SgxMutex<BTreeMap<String, Option<Arc<Listener>>>>,
+    abstr: SgxMutex<BTreeMap<String, Option<Arc<Listener>>>>,
+}
```

Comment:
As I have pointed out in my last round of comments, it is more logical for the `Listener` (previously known as `Server`), which maintains the incoming connections to a listening socket, to be part of `Stream`, instead of that of a global data structure (here, `AddressSpace`).

Admittedly, being "more logical" is not a very strong argument. So, I want to give a much stronger reason: improving performance and avoiding deadlocks. Ideally, a scalable implementation should have this property: socket A connecting to socket B should have no performance impact on socket C connecting socket D. Unfortunately, the current implementation is NOT; it has a global bottleneck--- `ADDRESS_SPACE`. Each and every connect or accept operation must hold the lock of `ADDRESS_SPACE` (see `AddressSpace::get_listener_ref`). To make it worse, the connect or accept operation may block indefinitely, which means the lock on `ADDRESS_SPACE` may be held for a very long time. Obviously, this is unacceptable.